First pass adding basic prerequisites (#296)
UCL/TDMS: README.md

<details>
<summary><img src="https://github.com/EgoistDeveloper/operating-system-logos/blob/master/src/24x24/UBT.png"/> Linux prerequisite setup</summary>

Assuming you don't already have them, you'll need a C++ compiler, CMake, OpenMP and FFTW.

For Debian-based distributions this should be as simple as
```{sh}
$ sudo apt install git gcc cmake libfftw3-dev libgomp1
```

</details>

<details>
<summary><img src="https://github.com/EgoistDeveloper/operating-system-logos/blob/master/src/24x24/MAC.png"/> MacOS prerequisite setup</summary>

On MacOS you will need an x86 compiler with libraries for OpenMP.
You'll need to download the latest [Xcode tools](https://apps.apple.com/app/xcode).

Everything else can be installed using [Homebrew](https://brew.sh):

```{sh}
$ brew install cmake fftw llvm
```

On an ARM Mac, you will need to install the x86 version of Homebrew:

```{sh}
$ arch -x86_64 zsh
$ arch -x86_64 /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
$ arch -x86_64 /usr/local/bin/brew install cmake fftw llvm
```

</details>

<details>

> <summary><img src="https://github.com/EgoistDeveloper/operating-system-logos/blob/master/src/24x24/WIN.png"/> Windows prerequisite setup</summary>

Author: Tagging @edggjames FYI...  👇

Everything else can be installed using [Homebrew](https://brew.sh):

```{sh}
$ brew install cmake fftw llvm
```

On an ARM Mac, you will need to install the x86 version of Homebrew:

```{sh}
$ arch -x86_64 zsh
$ arch -x86_64 /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
$ arch -x86_64 /usr/local/bin/brew install cmake fftw llvm
```

<!-- on MacOS it's just hdf5 as an argument to brew -->

</details>

<details>
<summary><img src="https://github.com/EgoistDeveloper/operating-system-logos/blob/master/src/24x24/WIN.png"/> Windows prerequisite setup</summary>

TDMS was developed on, and has been extensively tested on linux.
Support for Windows is quite new and experimental (please [report](https://github.com/UCL/TDMS/issues/new/choose) any issues you encounter!).

It might be more straightforward to use the [Windows subsystem for Linux (WSL2)](https://learn.microsoft.com/en-gb/windows/wsl/install), or set up an linux virtual machine.

However, TDMS _can_ be compiled natively on Windows.
This has been tested Windows 10 and 11, with PowerShell.

Assuming you don't already have them, you'll need to download and install:

> <!-- * HDF5 -->

Reviewer: ```suggestion
<!-- * TODO: When HDF5 comes in -->
```

Author: Huh. I had an even better comment with a link to the page... lost.

Everything else can be installed using [Homebrew](https://brew.sh):

```{sh}
$ brew install cmake fftw llvm
```

On an ARM Mac, you will need to install the x86 version of Homebrew:

```{sh}
$ arch -x86_64 zsh
$ arch -x86_64 /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
$ arch -x86_64 /usr/local/bin/brew install cmake fftw llvm
```

</details>

<details>
<summary><img src="https://github.com/EgoistDeveloper/operating-system-logos/blob/master/src/24x24/WIN.png"/> Windows prerequisite setup</summary>

TDMS was developed on, and has been extensively tested on linux.
Support for Windows is quite new and experimental (please [report](https://github.com/UCL/TDMS/issues/new/choose) any issues you encounter!).

It might be more straightforward to use the [Windows subsystem for Linux (WSL2)](https://learn.microsoft.com/en-gb/windows/wsl/install), or set up an linux virtual machine.

However, TDMS _can_ be compiled natively on Windows.
This has been tested Windows 10 with PowerShell.

Assuming you don't already have them, you'll need to download and install:

* [MATLAB](https://www.mathworks.com/products/matlab.html),

> * [Visual Studio](https://visualstudio.microsoft.com/vs/community/) and be sure to select the C++ kit,

Reviewer: ```suggestion
* [Visual Studio](https://visualstudio.microsoft.com/vs/community/) and **be sure to select the C++ kit**,
```

I feel like this should carry this emphasis since if you're like me (or the students that attend Carpentries workshops), you see a list of prerequisites and you don't look past the link you're given to the installer. Bold text helps draw the eye to "there's more to this step than you think".

* [CMake](https://cmake.org/download/),
* and [FFTW](https://www.fftw.org/install/windows.html).

Potentially the simplest way to get FFTW is via [conda](https://anaconda.org/conda-forge/fftw):

```{pwsh}
PS> conda install -c conda-forge fftw --yes
```

You'll need to ensure the paths to FFTW and MATLAB (the locations of `fftw3.dll` and `libmex.dll` respectively) are in the `env:Path`.

These can be found, e.g. by
```{pwsh}
PS> conda list fftw # assuming you installed via conda
PS> which.exe MATLAB
```

> Which should return something like `C:\Program Files (x86)\MATLAB\R20XXx\bin\matlab` and maybe `C:\ProgramData\envs\base\bin`.

Author: ```suggestion
Which should return something like `C:\Program Files (x86)\MATLAB\R20XXx\bin\` for MATLAB, and `C:\ProgramData\anaconda\envs\base\Library\lib` or `C:\Users\<your username>\anaconda\Library\lib\cmake\fftw3`.
```

If you downloaded FFTW and created `fftw3.dll` with `lib.exe`, you just need to know where you saved it.

You can append the paths:

```{pwsh}
PS> $env:Path += ";C:\Program Files (x86)\MATLAB\R20XXx\bin\;C:\ < wherever fftw3.dll is >"
```

Which will help Windows locate `.dll` files later.
For all following instructions, you'll have to substitute our mentions of `tdms` with `tdms.exe` and `$` is used to denote a command prompt which, in PowerShell would look like `PS>`

</details>

------

> You'll need to download and install [MATLAB](https://www.mathworks.com/products/matlab.html), and take note where the headers are installed.

Reviewer: If we're saying this, we might as well move it into the install list above (like when we say you need VSCode _and_ then do this other thing while you install it).

```md
Assuming you don't already have them, you'll need to download and install:

* [MATLAB](https://www.mathworks.com/products/matlab.html), and **take note where the headers are installed**
```

Author: Ah but this one is common to all three `<details>` sections. As in: do all of the things, and then get MATLAB.

Author: Perfectly happy to put it up into the sections though. We also say something similar _before_ all of the `<details>` blocks.

</details>

## Installation

To compile and install, follow these steps:

```{sh}
$ git clone git@github.com:UCL/TDMS.git
$ cd TDMS
$ git checkout v1.0.0 # the stable version
$ mkdir build; cd build
$ cmake ../tdms \
# -DMatlab_ROOT_DIR=/usr/local/MATLAB/R20XXx/ \
# -DFFTW_ROOT=/usr/local/fftw3/ \
# -DCMAKE_INSTALL_PREFIX=$HOME/.local/
$ cmake --build . --target install --config Release
```

If CMake cannot find MATLAB, FFTW, or install to the default installation prefix, uncomment the relevant line(s) and modify the path(s) accordingly.

You can check that `tdms` was installed correctly and is in your `PATH` by running:
```{sh}

> $ tdms -h

Author: `--help` actually isn't recognised (but perhaps it should be).

> $ tdms -h

Author: ```suggestion
$ tdms --help
```

$ which tdms
```

> Copy the full path (something like `/usr/local/bin/tdms`) into [`run_pstd_bscan.m`](https://github.com/UCL/TDMS/blob/main/examples/arc_01/run_pstd_bscan.m), replacing the ` 'tdms' ` text in the calls to the ``system()`` function.

Author: Something weird happened with your (good) suggestion on the [doxygen rendering](https://github-pages.ucl.ac.uk/TDMS/index.html#autotoc_md2). Perhaps this fixes...?
